osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,207 @@
+      {GENERIC_EXECUTE, "Generic Execute"},
+      {GENERIC_ALL, "Generic All"}};
+
+  for (auto const& perm : permVals) {
+    if ((pmask & perm.first) != 0) {
+      permList.push_back(perm.second);
+    }
+  }
+
+  return alg::join(permList, ",");
+}
+
+// helper function to convert inheritance type to string
+std::string inheritanceToStr(const unsigned long i) {
+  switch (i) {
+  case CONTAINER_INHERIT_ACE: // equivalent to SUB_CONTAINERS_ONLY_INHERIT
+    return "Container Inherit Ace";
+  case INHERIT_NO_PROPAGATE: // equivalent to NO_PROPAGATE_INHERIT_ACE
+    return "Inherit No Propagate";
+  case INHERIT_ONLY: // equivalent to INHERIT_ONLY_ACE
+    return "Inherit Only";
+  case OBJECT_INHERIT_ACE: // equivalent to SUB_OBJECTS_ONLY_INHERIT
+    return "Object Inherit Ace";
+  case SUB_CONTAINERS_AND_OBJECTS_INHERIT:
+    return "Sub containers and Objects Inherit";
+  case NO_INHERITANCE:
+    return "No Inheritence";
+  default:
+    return "Other";
+  }
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& t) {
+  char name[256];
+  char domain[256];
+  unsigned long size = 256; // Max username length for Windows
+  SID_NAME_USE accountType;
+
+  switch (t.TrusteeForm) {
+  case TRUSTEE_IS_SID:
+    // get the name from the SID
+    if (!LookupAccountSid(nullptr,
+                          reinterpret_cast<PSID>(t.ptstrName),
+                          name,
+                          &size,
+                          domain,
+                          &size,
+                          &accountType)) {
+      TLOG << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  case TRUSTEE_IS_NAME:
+    // get the name from ptstrName
+    return t.ptstrName;
+  case TRUSTEE_BAD_FORM:
+    // invalid
+    return "Invalid";
+  case TRUSTEE_IS_OBJECTS_AND_SID:
+    // ptstrName member is a pointer to an OBJECTS_AND_SID struct
+    if (!LookupAccountSid(
+            nullptr,
+            reinterpret_cast<PSID>(
+                reinterpret_cast<OBJECTS_AND_SID*>(t.ptstrName)->pSid),
+            name,
+            &size,
+            domain,
+            &size,
+            &accountType)) {
+      // printf("LookupAccountSid error: %u\n", GetLastError());
+      TLOG << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  case TRUSTEE_IS_OBJECTS_AND_NAME:
+    // ptstrName member is a pointer to an OBJECTS_AND_NAME struct
+    return reinterpret_cast<OBJECTS_AND_NAME_*>(t.ptstrName)->ptstrName;
+  default:
+    return "";
+  }
+}
+
+QueryData genNTFSACLPerms(QueryContext& context) {
+  QueryData results;
+  unsigned long result = 0;
+  PACL DACL = nullptr;
+  PEXPLICIT_ACCESS aceList = nullptr, aceItem = nullptr;
+  unsigned long aceCount = 0, aceIndex = 0;
+
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  for (const auto& path_string : paths) {
+    if (fs::exists(path_string)) {
```

Comment:
Rather than check if `fs::exists`, you could flip you'r logic here and save a level of scope:
```
if (!fs::exists(path_string)) { 
  continue;
}
// Do Logic Here.
```